modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/MapQueryResults.java

```diff
@@ -198,7 +199,16 @@ private void onAllClosed() {
         if (lazy)
             releaseQueryContext();
 
-        U.closeQuiet(qctx.queryMemoryTracker());
+        H2MemoryTracker tracker = qctx.queryMemoryTracker();
+
+        if (tracker != null) {
+            try {
+                tracker.close();
+            }
+            catch (Exception e) {
+                throw new IgniteException(e);
```

Comment:
why you rewrote it from closeQuiet to throw IgniteException?

Reply:
This is because we silently swallowed all errors if they were. Even assertion happened when we release more memory than we reserved. So we hid problems and bugs in our code.